### PR TITLE
Update package description and tags

### PR DESCRIPTION
**Pull request by bricelam**
cc @natemcmaster @rowanmiller @Eilon

### CODE REVIEW

File: src/Microsoft.Data.Sqlite/project.json [hunk 1]
```diff
@@ -1,6 +1,13 @@
 {
   "version": "1.0.0-*",
-  "description": "SQLite implementation of the System.Data.Common provider model.",
+  "description": "Provides a data provider for SQLite based on the System.Data.Common contract.\r\n\r\nCommonly Used Types:\r\nMicrosoft.Data.Sqlite.SqliteCommand\r\nMicrosoft.Data.Sqlite.SqliteConnection\r\nMicrosoft.Data.Sqlite.SqliteConnectionStringBuilder\r\nMicrosoft.Data.Sqlite.SqliteDataReader\r\nMicrosoft.Data.Sqlite.SqliteException\r\nMicrosoft.Data.Sqlite.SqliteFactory\r\nMicrosoft.Data.Sqlite.SqliteParameter\r\nMicrosoft.Data.Sqlite.SqliteTransaction",
+  "packOptions": {
+    "tags": [
+      "SQLite",
+      "Data",
+      "ADO.NET"
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/project.json]: ADO.NET Core? :trollface:

File: src/Microsoft.Data.Sqlite/project.json [hunk 2]
```diff
@@ -1,6 +1,13 @@
 {
   "version": "1.0.0-*",
-  "description": "SQLite implementation of the System.Data.Common provider model.",
+  "description": "Provides a data provider for SQLite based on the System.Data.Common contract.\r\n\r\nCommonly Used Types:\r\nMicrosoft.Data.Sqlite.SqliteCommand\r\nMicrosoft.Data.Sqlite.SqliteConnection\r\nMicrosoft.Data.Sqlite.SqliteConnectionStringBuilder\r\nMicrosoft.Data.Sqlite.SqliteDataReader\r\nMicrosoft.Data.Sqlite.SqliteException\r\nMicrosoft.Data.Sqlite.SqliteFactory\r\nMicrosoft.Data.Sqlite.SqliteParameter\r\nMicrosoft.Data.Sqlite.SqliteTransaction",
```
Comment (rowanmiller) [inline src/Microsoft.Data.Sqlite/project.json]: I think I prefer the old "SQLite implementation of the System.Data.Common provider model" wording, what do others think?

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/project.json]: No preference here. I was just trying to adopt the phrasing used by some of the corefx packages.

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/project.json]: Agreeing with @rowanmiller on the old wording, but I like that "common types" section.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/project.json]: So you're saying "Provides a ... provider" is weird? :smirk: (I copied it from SqlClient)

Comment (rowanmiller) [inline src/Microsoft.Data.Sqlite/project.json]: Hmmm, perhaps we could use something like...

> Provides a provider that provides SQLite data access provided you are ok with not being provided with typed data set and other provider things not provided by System.Data.Common

...on a serious note, my vote is to keep the old wording and just add the "Commonly used types" part

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/project.json]: Done.

File: src/Microsoft.Data.Sqlite/project.json [hunk 3]
```diff
@@ -9,10 +16,7 @@
     ],
     "xmlDoc": true,
     "embed": {
-      "include": [
-        "Properties/Microsoft.Data.Sqlite.rd.xml",
-        "Strings.resx"
```
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/project.json]: Assuming this means resx files are automatically embedded now?

**Comment by natemcmaster** [comment]
LGTM

File: src/Microsoft.Data.Sqlite/project.json (hunk 3 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/project.json]: Why does it need to be embedded? The compiler embeds the transformed `*.resources` file.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/project.json]: Discussed offline and we don't know why this was ever there in the first place. :blush:

**Comment by bricelam** [comment]
I also did a pass of code cleanup and added it to this PR since they're also non-functional changes.

File: src/Microsoft.Data.Sqlite/project.json (hunk 3 above)
Comment (Eilon) [inline src/Microsoft.Data.Sqlite/project.json]: Yup, this was definitely just always wrong.

**Pull request merged**